tests/lean/derivingRpcEncoding.lean.expected.out

```diff
@@ -0,0 +1,41 @@
+instRpcEncodingWithRpcRefFooRefRpcRef : RpcEncoding (WithRpcRef FooRef) Lsp.RpcRef
+ok: {"p": "0"}
+protected inductive Bar.RpcEncodingPacket : Type
+number of parameters: 0
+constructors:
+Bar.RpcEncodingPacket.mk : Lsp.RpcRef → FooJson → Bar.RpcEncodingPacket
+instRpcEncodingBarRpcEncodingPacket : RpcEncoding Bar Bar.RpcEncodingPacket
+ok: {"fooRef": {"p": "0"}, "fooJson": {"s": ""}}
+instRpcEncodingBarTransRpcEncodingPacket : RpcEncoding BarTrans BarTrans.RpcEncodingPacket
+ok: {"bar": {"fooRef": {"p": "0"}, "fooJson": {"s": ""}}}
+instRpcEncodingBazRpcEncodingPacket : RpcEncoding Baz Baz.RpcEncodingPacket
+ok: {"arr": []}
+protected inductive FooGeneric.RpcEncodingPacket : Type → Type
+number of parameters: 1
+constructors:
+FooGeneric.RpcEncodingPacket.mk : {αPacket : Type} → αPacket → Option αPacket → FooGeneric.RpcEncodingPacket αPacket
+instRpcEncodingFooGenericRpcEncodingPacket : (α αPacket : Type) →
+  [inst : RpcEncoding α αPacket] → RpcEncoding (FooGeneric α) (FooGeneric.RpcEncodingPacket αPacket)
+ok: {"a": 0}
+ok: {"b": 42, "a": 3}
+instRpcEncodingBazInductiveRpcEncodingPacket : RpcEncoding BazInductive BazInductive.RpcEncodingPacket
+ok: {"baz":
+ [{"fooRef": {"p": "0"}, "fooJson": {"s": ""}},
+  {"fooRef": {"p": "1"}, "fooJson": {"s": ""}}]}
+inductive FooInductive.RpcEncodingPacket : Type → Type
+number of parameters: 1
+constructors:
+FooInductive.RpcEncodingPacket.a : {αPacket : Type} → αPacket → Lsp.RpcRef → FooInductive.RpcEncodingPacket αPacket
+FooInductive.RpcEncodingPacket.b : {αPacket : Type} → Nat → αPacket → Nat → FooInductive.RpcEncodingPacket αPacket
+instRpcEncodingFooInductiveRpcEncodingPacket : (α αPacket : Type) →
+  [inst : RpcEncoding α αPacket] → RpcEncoding (FooInductive α) (FooInductive.RpcEncodingPacket αPacket)
+ok: {"a": [{"baz": []}, {"p": "0"}]}
+ok: {"b": [42, {"baz": []}, 0]}
```

Comment:
@Vtec234 Reading the comment in the derive handler correctly, the constructor `| b : (n : Nat) → (a : α) → (m : Nat) → FooInductive α` should produce a map instead of an array:
```suggestion
ok: {"b": {"n": 42, "a": {"baz": []}, "m": 0}}
```

However this has already been broken on master for quite some time I believe.  In any case this test will catch future changes in the serialization format.

Reply:
Indeed that is what the `deriving` was supposed to do, and indeed I just confirmed that it does not work. It is just `ToJson` which encodes `|  c (n m : Nat) : FooInductive α` as `{"c": {"n": 1, "m": 2}}`, but `b` as the test output. Could you add a test checking that the `c` variant works?

Reply:
Okay, fixed.  The test was already there (but obviously produces a different output now).